keras/callbacks.py

```diff
@@ -779,32 +785,46 @@ def is_indexed_slices(grad):
         else:
             self.writer = tf.summary.FileWriter(self.log_dir)
 
-        if self.embeddings_freq:
+        if self.embeddings_freq and self.embeddings_data is not None:
```

Comment:
Let's say we only want to visualize the weights, should we always pass data?

Reply:
This PR's intention is to fix the callback from only visualizing the weights to actually embedding the data using the weights. Visualizing the weights won't be possible with these changes.

Reply:
Ah, I guess we should be able to do both. Removing this feature could impact other users. @chleibig @fchollet any thoughts on that?

Reply:
I wouldn't miss the original functionality, but that's just my personal opinion...revising the conversation of the original PR that had introduced the weight embedding functionality might reveal whether this is needed or indeed was unintentionally introduced.